src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -623,17 +623,13 @@ public void visitToken(DetailAST ast) {
             final boolean isStaticAndNotLastImport = isStatic && !lastImportStatic;
             doVisitToken(ident, isStatic, isStaticAndNotLastImport, line);
         }
-        else if (option == ImportOrderOption.BOTTOM || option == ImportOrderOption.UNDER) {
-            final boolean isLastImportAndNonStatic = lastImportStatic && !isStatic;
-            doVisitToken(ident, isStatic, isLastImportAndNonStatic, line);
-        }
         else if (option == ImportOrderOption.INFLOW) {
             // "previous" argument is useless here
             doVisitToken(ident, isStatic, true, line);
         }
         else {
-            throw new IllegalStateException(
-                    "Unexpected option for static imports: " + option);
```

Comment:
This throw isn't needed because option can never be set to illegal value as it is controlled by the enum in the setter method.